pds4indextools/pds4_create_xml_index.py

```diff
@@ -222,40 +217,190 @@ def process_schema_location(file_path):
     return xsd_urls
 
 
-def process_tags(xml_results, key, root, namespaces, prefixes, args):
-    """Process XML tags based on the provided options.
+def process_headers(label_results, key, root, namespaces, prefixes):
+    """Process headers to have more readable contents. 
 
-    If the --xpaths command is used, the XPath is converted into a format that
-    contains the names and namespaces of all the parent elements of that element.
-    If the --xpaths command is not used, the XPath is converted into the
-    associated element tag of that element, and given its associated namespace. These
-    values then replace their old versions in the xml_results dictionary.
+    Processes XPath headers by converting parts of the XPath into element tags,
+    replacing namespaces with prefixes, and updating the label_results dictionary.
+    If a duplicate XPath is encountered, it appends an underscore and a number
+    to make the XPath unique.
```

Comment:
Oh maybe this answers my question below about where the `_1` comes from in the SPICE kernel list. I don't understand how this can happen. Why is there a case where you get `_1` instead of `<1>`?

Reply:
So, the issue lies not in the predicates (turns out that's what the `[1]` are called in these), but in the namespaces that are used. The `geom` namespace does not use predicates. And since these numbers don't exist until they are artificially added, nothing stops them from overwriting each other. The added underscore and number are just to keep everything in order until the XPaths get the predicates and go through renumbering.